src/transformers/trainer_pt_utils.py

```diff
@@ -360,3 +361,32 @@ def finalize(self):
         if self._offsets[0] != self.process_length:
             logger.warn("Not all data has been set. Are you sure you passed all values?")
         return nested_truncate(self._storage, self.num_samples)
+
+
+@dataclass
+class LabelSmoother:
+    """
+    Adds the label-smoothing on a pre-computed output from a Transformers models.
+
+    Args:
+        epsilon (:obj:`float`, `optional`, defaults to 0.1):
+            The label smoothing factor.
+        ignore_index (:obj:`int`, `optional`, defaults to -100):
+            The index in the labels to ignore when computing the loss. Set to :obj:`None` to ignore none of the
```

Comment:
(question) Not sure if the `Set to :obj:`None` to ignore none of the indices` is really required. For me the default case should be `-100` but if the user does not want to ignore any indices, there should simply be no -100 in the `labels` no? Or is there a case where one would have -100 in the labels **and** wants to apply `label_smoothing` on those?

Reply:
Yeah, the docstring is just badly worded.